### PR TITLE
feat(LERN-79): Implemented role based access control

### PR DESCRIPTION
**Pull request by A1eeee**

### CODE REVIEW

File: backend/src/main/kotlin/ch/learntrack/backend/security/LearnTrackUserDetails.kt [hunk 1]
```diff
@@ -2,10 +2,15 @@ package ch.learntrack.backend.security
...
 import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.authority.SimpleGrantedAuthority
 import org.springframework.security.core.userdetails.UserDetails
 
+private const val ROLE_PREFIX = "ROLE_"
+
 public class LearnTrackUserDetails(public val user: User) : UserDetails {
-    override fun getAuthorities(): List<GrantedAuthority> = listOf()
+    override fun getAuthorities(): List<GrantedAuthority> = listOf(SimpleGrantedAuthority(
+        ROLE_PREFIX + user.userRole.toString(),
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/security/LearnTrackUserDetails.kt:12]: Maybe we could use user.userRole.name here to?

File: backend/src/test/kotlin/ch/learntrack/backend/admin/grade/GradeIntegrationTest.kt [hunk 2]
```diff
@@ -66,6 +67,7 @@ class GradeIntegrationTest: IntegrationTest() {
                     .queryParam("schoolId", schoolTemplateId)
                     .build()
             }
+            .headers { headers -> headers.setBearerAuth(tokenService.createJwtToken(userTemplateId)) }
```
Comment (deepspace1000) [inline backend/src/test/kotlin/ch/learntrack/backend/admin/grade/GradeIntegrationTest.kt:70]: maybe some refactoring we could do in this PR... remove the default jwt header in the IntegrationTest file and also add 2 more users to the userTemplate: 
`createAdminUserFromTemplate`, `createTeacherUserFromTemplate` and `createStudentUserFromTemplate`.

Comment (deepspace1000) [reply backend/src/test/kotlin/ch/learntrack/backend/admin/grade/GradeIntegrationTest.kt:70]: so in all tests we don't want to be authenticated like the registration or login and stuff we are by default unauthenticated and if we need to be we set the BearAuth in each test like here.

**Review by deepspace1000 — COMMENTED** [review]
Very nice. just some additional suggestions for the PR.

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Review by deepspace1000 — COMMENTED** [review]
Go through all test again and remove whats unnecessary.

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Review by deepspace1000 — APPROVED** [review]
nice

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Pull request merged**